crates/cli-tools/src/action.rs

```diff
@@ -370,12 +373,15 @@ pub fn optimize_wasm(applet: impl AsRef<Path>, opt_level: Option<OptLevel>) -> R
     opt.arg(applet.as_ref());
     opt.arg("-o");
     opt.arg(applet.as_ref());
-    cmd::execute(&mut opt)?;
+    cmd::execute(&mut opt).await?;
     Ok(())
 }
 
-fn metadata(dir: impl Into<PathBuf>) -> Result<Metadata> {
-    let metadata = MetadataCommand::new().current_dir(dir).no_deps().exec()?;
+async fn metadata(dir: impl Into<PathBuf>) -> Result<Metadata> {
+    let dir = dir.into();
+    let metadata =
+        tokio::task::spawn_blocking(|| MetadataCommand::new().current_dir(dir).no_deps().exec())
```

Comment:
Maybe we could introduce a helper for this. It seems common.

Reply:
It's actually not that useful.